### PR TITLE
fix!: nullability & other things

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

File: DisCatSharp.ApplicationCommands/Checks/ApplicationCommandEqualityChecks.cs [hunk 1]
```diff
@@ -295,37 +226,21 @@ internal static bool DeepEqual(DiscordApplicationCommand source, DiscordApplicat
...
 				var sOpt = JsonConvert.SerializeObject(minimalSourceOptions, Formatting.None);
 				var tOpt = JsonConvert.SerializeObject(minimalTargetOptions, Formatting.None);
-				
+
 				eqCheck2 = rootCheck && sOpt == tOpt;
 			}
 
 			return eqCheck1 && eqCheck2;
 		}
+		// ReSharper disable once RedundantIfElseBlock
```
Comment (xMaxximum) [inline DisCatSharp.ApplicationCommands/Checks/ApplicationCommandEqualityChecks.cs:235]: why does this get marked as a redundant if else block?

File: DisCatSharp.ApplicationCommands/ExtensionMethods.cs [hunk 2]
```diff
@@ -69,10 +69,7 @@ public static ApplicationCommandsExtension GetApplicationCommands(this DiscordCl
 	public static async Task<IReadOnlyDictionary<int, ApplicationCommandsExtension>> GetApplicationCommandsAsync(this DiscordShardedClient client)
 	{
 		await client.InitializeShardsAsync().ConfigureAwait(false);
-		var modules = new Dictionary<int, ApplicationCommandsExtension>();
-		foreach (var shard in client.ShardClients.Values)
-			modules.Add(shard.ShardId, shard.GetExtension<ApplicationCommandsExtension>());
-		return modules;
+		return client.ShardClients.Values.ToDictionary(shard => shard.ShardId, shard => shard.GetExtension<ApplicationCommandsExtension>());
```
Comment (xMaxximum) [inline DisCatSharp.ApplicationCommands/ExtensionMethods.cs:72]: is there maybe any other way besides using reflection utilities or are the performance impacts by it not significant enough?

File: DisCatSharp.Common/Utilities/RuntimeInformation.cs [hunk 3]
```diff
@@ -25,6 +25,7 @@
 using System.Linq;
 using System.Reflection;
 using System.Text;
+// ReSharper disable PossibleNullReferenceException
```
Comment (xMaxximum) [inline DisCatSharp.Common/Utilities/RuntimeInformation.cs:28]: "fix!: nullability" 🤔

**Review by xMaxximum — DISMISSED** [review]
LGTM after looking at the changes

**Comment by Lulalaby** [comment]
@xMaxximum this is still a draft XD
I just have enough of C# the weekend xD

**Pull request closed without merging**